pyccel/parser/semantic.py

```diff
@@ -1703,6 +1707,102 @@ def _visit_Max(self, expr, **settings):
         args = self._visit(expr.args, **settings)
         return Max(*args)
 
+    def _visit_lhs_Assign(self, lhs, d_var, rhs, **settings):
```

Comment:
I like the logic implemented in this function, but I think we should change its name.

Every other method in class `SemanticParser` follows the naming convention `_visit_TypeName` where `TypeName` is indeed the name of the type of the argument. But we do not have a type `lhs_Assign`, so I think this should be called differently.